scripts/k3s-versions.go

```diff
@@ -27,8 +32,12 @@ type Channel struct {
 
 // getK3sChannels returns a map of all non-prerelease channels, plus "latest" and "stable".
 // The values are the latest release for each channel.
-func getK3sChannels() (map[string]string, error) {
-	resp, err := http.Get("https://update.k3s.io/v1-release/channels")
+func getK3sChannels(ctx context.Context) (map[string]string, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, k3sChannelsEndpoint, http.NoBody)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create HTTP request: %w", err)
+	}
+	resp, err := http.DefaultClient.Do(req)
```

Comment:
This does not exactly make the code easier to read than a simple `http.Get()`, but I understand the value of consistency...

Speaking of consistency, we should use `http.DefaultClient` in the rest of the code too, instead of `client := &http.Client{}`.